core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java

```diff
@@ -842,11 +842,27 @@ public String toString(ConnectorSession session)
 
     private String formatRanges(ConnectorSession session, int limit)
     {
-        return Stream.concat(
-                IntStream.range(0, min(getRangeCount(), limit))
+        if (isNone()) {
+            return "{}";
+        }
+        if (getRangeCount() == 1) {
+            return "{" + getRangeView(0).formatRange(session) + "}";
+        }
+        if (limit < 2) {
+            return format("{%s, ...}", getRangeView(0).formatRange(session));
+        }
+        // Print first (limit - 1) elements, followed by last element
```

Comment:
Do we print `...` as last element? Is it `{1, 2, ... N}` or `{1, 2, N, ...}`? The latter is preferred

Reply:
`...` is last element only when limit=1 and there is more than 1 element as there is no other choice for this case.
Otherwise we do print `{1, 2, ... N}`